pkg/instrumentation/sdk.go

```diff
@@ -661,6 +648,20 @@ func getIndexOfEnv(envs []corev1.EnvVar, name string) int {
 	return -1
 }
 
+func appendIfNotSet(envs []corev1.EnvVar, newEnvVars ...corev1.EnvVar) []corev1.EnvVar {
+	keys := make(map[string]struct{}, len(envs))
+	for _, e := range envs {
+		keys[e.Name] = struct{}{}
+	}
+	for _, envVar := range newEnvVars {
+		if _, ok := keys[envVar.Name]; !ok {
+			envs = append(envs, envVar)
+			keys[envVar.Name] = struct{}{}
+		}
+	}
+	return envs
+}
```

Comment:
Could you add a few simple tests for this function?

Reply:
Good idea, looks like I have a regression too according to CI. Will work on it more.